src/OpenTelemetry.Api/Trace/TelemetrySpan.cs

```diff
@@ -47,7 +47,11 @@ public ActivitySpanId ParentSpanId
     /// <param name="value">Status to be set.</param>
     public void SetStatus(Status value)
     {
+#pragma warning disable
+        // TODO: Remove this when SetStatus is deprecated
         this.Activity?.SetStatus(value);
+#pragma warning disable
+        this.Activity?.SetStatus((ActivityStatusCode)value.StatusCode);
```

Comment:
not sure why we do this both ways?

Reply:
this is needed for backward compatibility. if we simply remove the older way, we might break code/libraries still relying on older way.